src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -69,23 +99,19 @@ - (void)runLoop
         unsigned int delaySeconds = (10 * 60); // 10 minutes.
         DDLogDebug(
             @"%@ No more expiring messages. Setting next check %u seconds into the future", self.tag, delaySeconds);
-        [self runBy:now + delaySeconds * 1000];
+        [self runByDate:[NSDate ows_dateWithMillisecondsSince1970:now + delaySeconds * 1000]];
         return;
     }
 
     uint64_t nextExpirationAt = [nextExpirationTimestampNumber unsignedLongLongValue];
-    uint64_t runByMilliseconds;
-    if (nextExpirationAt < now + 1000) {
-        DDLogWarn(@"%@ Next run requested at %llu, which is too soon. Delaying by 1 sec to prevent churn",
```

Comment:
`runByDate` already ensures that we don't run more often than once per second, so this logic is no longer necessary.